Document filters as processors (#1605)
fluent/fluent-bit-docs: pipeline/processors/filters.md

> # Filters

Reviewer: ```suggestion
# Filters as processors
```

Any Fluent Bit [filter](../filters/README.md) can be used as a processor.

> ## grep example
> 
> In this example the grep filter is used as an output processor to send Log records only if they match a regular expression.

Reviewer: ```suggestion
## Grep example

In this example, the [Grep](../filters/grep) filter is used as an output processor that sends log records only if they match a specified regular expression.
```

> {% tabs %}
> {% tab title="fluent-bit.yaml" %}
> 
> ```yaml
> pipeline:
>   inputs:
>     - name: tail
>       path: lines.txt
>       parser: json
>   outputs:
>     - name: stdout
>       match: '*'
>       processors:
>         logs:
>           - name: grep
>             regex: log aa
> ```
> 
> {% endtab %}
> {% endtabs %}

Reviewer: ````suggestion
{% code title="fluent-bit.yaml" %}
```yaml
pipeline:
  inputs:
    - name: tail
      path: lines.txt
      parser: json
  outputs:
    - name: stdout
      match: '*'
      processors:
        logs:
          - name: grep
            regex: log aa
```
{% endcode %}
````

I _just_ learned today that there's a dedicated setting to add file names to code blocks... but using a tab is a clever workaround 😆
